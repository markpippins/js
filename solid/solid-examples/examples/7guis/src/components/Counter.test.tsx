import { afterEach, beforeEach, describe, expect, test } from 'vitest';
import { render, fireEvent } from 'solid-testing-library';
import Counter from './Counter';
import type { Result } from 'solid-testing-library/dist/types';

describe('<Counter />', () => {
  let result: Result;
  beforeEach(() => {
    result = render(() => <Counter />);
  });
  afterEach(() => result.unmount());

  test('renders', () => {
    const { container } = result;
    expect(container).toMatchSnapshot();
  });

  test('increments value', async () => {
    const { queryByRole, getByText } = result;
    const button = (await queryByRole('button')) as HTMLButtonElement;
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(/Increment Count/);
    const display = (await getByText(/Clicks/)) as HTMLSpanElement;
    expect(display).toHaveTextContent(/Clicks: 0/);
    fireEvent.click(button);
    expect(display).toHaveTextContent(/Clicks: 1/);
  });
});
