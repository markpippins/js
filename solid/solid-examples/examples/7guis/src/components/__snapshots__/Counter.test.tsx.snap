// Vitest Snapshot v1

exports[`<Counter /> > renders 1`] = `
<div>
  <button
    class="_increment_l2pqx_5"
    type="button"
  >
    Increment Count
  </button>
  <div
    class="_countContainer_l2pqx_1"
  >
    <span
      class="_count_l2pqx_1"
    >
      Clicks: 
      <!--#-->
      0
      <!--/-->
    </span>
  </div>
</div>
`;
